src/backend/executor/nodeSeqscan.c

```diff
@@ -87,8 +101,14 @@ SeqNext(SeqScanState *node)
 	/*
 	 * get the next tuple from the table
 	 */
-	if (table_scan_getnextslot(scandesc, direction, slot))
+	while (table_scan_getnextslot(scandesc, direction, slot))
+	{
+		if (node->filter_in_seqscan && node->filters &&
```

Comment:
```c++
if (!node->filter_in_seqscan ||  !node->filters)
{
     if (table_scan_getnextslot(scandesc, direction, slot))
         return slot;
}
else 
{
     while (table_scan_getnextslot(scandesc, direction, slot))
     {
            .....
     }
}
```

this make origin path more efficient and readable ?

Reply:
good idea! I fix in https://github.com/apache/cloudberry/pull/724/commits/bcf93e66f72a35db0ab5694a92c59c768423b1ab